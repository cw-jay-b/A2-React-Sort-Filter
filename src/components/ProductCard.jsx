import React from 'react';
import '../styles/ProductCard.css';
import NoCarImg from "../assets/default-car.svg"; 

const ProductCard = (product) => {
  const fuelMap = ['Petrol', 'Diesel', 'CNG', 'LPG', 'Electric', 'Hybrid'];
  // console.log(product);
  
  return (
    <div className="product-card">
      {/* {product.profileId} */}
      <img src={product.stockImages[0] || NoCarImg } alt={product.carName} className="product-image" />
      <h2 className="product-title">{product.carName}</h2>
      <p className="product-details">
        <span className='km'>{product.km} km </span> |
        <span className='fuel'>{product.fuel} </span> |
        <span className='cityname'>{product.cityName}</span>
      </p>
      <p className="product-price">Rs. {product.price}</p>
      <button className="product-button">Get Seller Details</button>
    </div>
  );
};

export default ProductCard;
