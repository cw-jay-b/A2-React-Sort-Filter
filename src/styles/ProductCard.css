
.products-wrapper {
  margin-top: 1rem;
  width: 37rem;
  gap: 1rem;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: flex-start;
}

.product-card {
  display: flex;
  flex-direction: column;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 8px;
  box-shadow: 0 2px 4px lightgray;
  width: 18rem;
  padding: 15px;
  box-sizing: border-box;
  justify-content: space-between;
}
  
  .product-image {
    height: 193.17px;
    width: 100%;
    height: auto;
  }
  
  .product-title {
    font-size: 1rem;
    font-weight: 600;
    margin: 10px 0;
  }

  .product-details {
    display: flex;
    flex-wrap: nowrap;
    font-size: 13px;
    column-gap: 4px;
    color: #555;
  }
  
  /* .km { 
    border-right: 1px solid gray;
  }
  .fuel { 
    grid-column: 2; 
    border-right: 1px solid gray;
  }
  .cityname { 
    grid-column: 3; 
  } */

  
  
  .product-price {
    font-size: 18px;
    font-weight: 700;
    margin: 10px 0;
    color: #000;
  }
  
  .product-button {
    width: 100%;
    background-color: #e03012;
    color: #fff;
    border: none;
    border-radius: 4px;
    padding: 10px;
    cursor: pointer;
    transition: background-color 0.3s;
    font-size: 15px;
    font-weight: 700;
  }
  
  .product-button:hover {
    background-color: darkred;
  }
  