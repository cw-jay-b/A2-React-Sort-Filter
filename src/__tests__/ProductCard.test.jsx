import React from 'react';
// import { shallow } from 'enzyme';
import { render, screen } from '@testing-library/react';
import { describe, test, expect,  } from 'vitest';
import '@testing-library/jest-dom';

import ProductCard from '../components/ProductCard';

describe('ProductCard', () => {
  const dummyCarDetails = {
    carName: 'Maruti Suzuki Wagon R LXi',
    price: '2.64 Lakh',
    km: '62,000',
    fuel: 'Petrol',
    cityName: 'Mumbai',
    stockImages: []
  };

  test('renders car details correctly', () => {
    const productCard = render(<ProductCard {...dummyCarDetails} />);
    expect(productCard).toMatchSnapshot(); // Snapshot testing
    
    expect(productCard.getByText(dummyCarDetails.carName)).toBeInTheDocument();
    expect(productCard.getByText(dummyCarDetails.price)).toBeInTheDocument();
    expect(productCard.getByText(dummyCarDetails.km)).toBeInTheDocument();
    expect(productCard.getByText(dummyCarDetails.fuel)).toBeInTheDocument();
    expect(productCard.getByText(dummyCarDetails.cityName)).toBeInTheDocument();
  });
});
