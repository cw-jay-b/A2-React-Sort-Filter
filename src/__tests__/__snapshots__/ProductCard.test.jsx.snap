// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductCard > renders car details correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="product-card"
      >
        <img
          alt="Maruti Suzuki Wagon R LXi"
          class="product-image"
          src="/src/assets/default-car.svg"
        />
        <h2
          class="product-title"
        >
          Maruti Suzuki Wagon R LXi
        </h2>
        <p
          class="product-details"
        >
          <span
            class="km"
          >
            62,000
             
          </span>
           km |
          <span
            class="fuel"
          >
            Petrol
             
          </span>
           |
          <span
            class="cityname"
          >
            Mumbai
          </span>
        </p>
        <p
          class="product-price"
        >
          2.64 Lakh
        </p>
        <button
          class="product-button"
        >
          Get Seller Details
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="product-card"
    >
      <img
        alt="Maruti Suzuki Wagon R LXi"
        class="product-image"
        src="/src/assets/default-car.svg"
      />
      <h2
        class="product-title"
      >
        Maruti Suzuki Wagon R LXi
      </h2>
      <p
        class="product-details"
      >
        <span
          class="km"
        >
          62,000
           
        </span>
         km |
        <span
          class="fuel"
        >
          Petrol
           
        </span>
         |
        <span
          class="cityname"
        >
          Mumbai
        </span>
      </p>
      <p
        class="product-price"
      >
        2.64 Lakh
      </p>
      <button
        class="product-button"
      >
        Get Seller Details
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
